.changeset/odd-seahorses-learn.md

```diff
@@ -0,0 +1,4 @@
+---
+---
+
+chore: adding type checking
```

Comment:
A couple of questions:
1. Why did you apply these only to a subset of packages?
2. Why do some packages call the validation during `build` and others during `test`?
3. Wouldn't it be possible to add this logic into the `postbuild` script since it applies to all packages already?
4. How much slower did our routines get by adding this validation?

Reply:
1. I have applied type checking on the `package.json` level for packages that don't run the `postbuild` command.
2. I used the `pretest` hook for `docs-snippets` and `fuel-gauge` only. I assumed that post-pretest stage would be where type errors would arise.
3. I have added it to the [`scripts/postbuild.ts`](https://github.com/FuelLabs/fuels-ts/pull/3046/files#diff-21462feba05a79008c54d9a3267569b27b8897540ac86e47110e57f25402d224).
4. Comparing `test` workflows in CI - it appears there is about 45 seconds difference. Maybe moving this to the type checking to the `lint` step might be beneficial.

Reply:
Great, fine by me. Thanks for the answers.